### ISSUE TITLE
No way to sample a `texture_1d` from a vertex shader

### ISSUE THREAD

**Issue by Tim-Evans-Seequent** [gpuweb/gpuweb#5001]
`textureSample(texture_1d, ...)` isn't allowed in vertex shaders, and `textureSampleLevel(texture_1d, ...)` isn't allowed at all. This appears to leave us with no way to sample a 1D texture from a vertex shader.

I think that a reasonable solution to this would be to allow `textureSampleLevel(texture_1d, ...)` to match 2D textures. For what it's worth `wgpu` currently allows it, and it seems to work there.

**Comment by kainino0x** [comment]
Interesting, thank you for the report! I'm surprised no one has found this earlier, it seems like a pretty major gap.

This seems like an oversight, unless I'm missing some reason we didn't include this in the first place. A quick search turns up the rejected PR #1814 which sort of explains why we don't have 1d `textureSampleLevel`, but then we added 1d `textureDimensions` (#1938) without adding 1d `textureSampleLevel`.

It could sort of make sense to have 1d `textureSampleBase`, but for the same reason 1d `textureDimensions` takes a level even though it's clamped to 0, I think we should have 1d `textureSampleLevel`, not `textureSampleBase`.

(because 1d textures [don't have mipmapping](https://gpuweb.github.io/gpuweb/#dom-gputexturedimension-1d))

In the meantime I guess you'll have to use 1d `textureLoad`, or use a 2d texture in place of your 1d texture.

**Comment by jimblandy** [comment]
Obviously, we'll need to make sure the underlying platform support is there, but if it is, then it seems we should just add this, once we know how we want to address the absence of mipmaps.

**Comment by kdashg** [comment]
<details><summary><a href="https://github.com/gpuweb/gpuweb/wiki/WGSL-2024‐12‐03-Minutes#no-way-to-sample-a-texture_1d-from-a-vertex-shader--issue-5001">WGSL 2024-12-03 Minutes</a></summary>

* DN: Started to do some archaeology here on how we ended up here
* JB: Seems like the main difference for 1d textures is that they don’t support mipmaps. `textureSampleLevel` would be weird because you don’t have mipmapping, so there’s only level=0. Seems like we should just add what we need.
* DN: Things I’m concerned about:
    * Found an issue four(?) years ago where you’d try it in GLSL and glslang would insert level=0 for vertex.
    * We did note in previous discussions that we expected users to mostly call textureSampleLevel(level=0) in vert shaders.
* DN: I’d like to expose what’s supported, so I think we need to have an investigation here. Concern about sampler addressing modes. Make sure they work.
* JB: We don’t want to deviate too much from what the platforms are doing, but we would like to give authors a hint to authors that we aren’t using derivatives here.
* JB: 
* DN: There is textureSampleBaseClampToEdge and that sounds like what we want to do here. Except it clamps the coordinates so the sampler addressing modes don’t kick in.
* KG: For external textures?
* DN: Both external and tex2d.
* KG: Do we have a textureSampleBase?
* DN: No
* KG: Are we missing textureSampleLevel(tex1d)?
* JB: Yes
* JP: We do support tex1d in textureDimensions(), and asking for OOB level is indeterminant value (also textureNumLevels)
* KG: Might be easiest then to just add textureSampleLevel(tex1d)?
* DN: If we add CTS and it says it works, that seems like the way to do it.
* **Resolved: Write CTS and Proposal**
* JP: Should this be a lang feature?
* Multiple: I don’t think so.

</details>

**Comment by kainino0x** [comment]
> * JP: Should this be a lang feature?
> * Multiple: I don’t think so

IIRC the purpose of language features is for feature-detection, so it makes sense to me to have a language feature. Not a huge deal though.
